### PR TITLE
Sw models

### PR DESCRIPTION
**Pull request by sydneywait**
# Description

Added models for:
Customer
PaymentType
EmployeeTraining
TrainingProgram
ProductType
Product
Order
OrderProduct

I also fixed the ProjectReference to this project (BangazonAPI) in the TestBangazonAPI.csproj file  It was showing StudentExercises.  Be sure to remove this in your local files if github automerges and keeps both!!

*****ETA--There is also now a boilerplate SQL file that will restore the database and add seed data.  It is on the same level as the project folders.

## Related Ticket(s)
Setting up the project models to begin work on the tickets.

## Steps to Test
Please provide thorough instructions for how to manually test the changes on this PR.

 Open the model files and individually check to see that they have Primary Keys.
Check that all data types are correct according to the ERD on the whiteboard
Check that the program compiles without errors

# Checklist:
- [ na] I have commented my code, particularly in hard-to-understand areas
- [ na] I have made corresponding changes to the documentation
- [x ] My changes generate no new errors
- [na ] I have added tests that prove my fix is effective or that my feature works
- [ x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by iamnickmitchell** [comment]
🍝

File: BangazonAPI/Models/Customer.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace BangazonAPI.Models
+{
+    public class Customer
+    {
+        public int Id { get; set; }
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
```
Comment (jordan-castelloe) [inline BangazonAPI/Models/Customer.cs:12]: According to ticket #1 when we display a customer we should have the capacity to also display a list of products they've created and payment types they've created. We should have two `List` properties on this model of type `Product` and `PaymentType` respectively

File: BangazonAPI/Models/EmployeeTraining.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+using System.Threading.Tasks;
+
+namespace BangazonAPI.Models
+{
+    public class EmployeeTraining
+    {
+
+        public int Id { get; set; }
+        public int EmployeeId { get; set; }
+        public int TrainingProgram { get; set; }
```
Comment (jordan-castelloe) [inline BangazonAPI/Models/EmployeeTraining.cs]: Do we ever need to expose an `EmployeeTraining` instance to the user? I can't think of a situation where we would, but feel free to correct me.

If we want to represent the many-to-many relationship between `Employees` and `TrainingProgram` we should have a list on Employee of type `TrainingProgram` and a list on `TrainingProgram` of type `Employee`

File: BangazonAPI/Models/Order.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace BangazonAPI.Models
+{
+    public class Order
+    {
+        public int Id { get; set; }
+        public int CustomerId { get; set; }
+        public int PaymentTypeId { get; set; }
```
Comment (jordan-castelloe) [inline BangazonAPI/Models/Order.cs:12]: Orders should have a list of `Products` and a list of `Customers` based on the requirements in ticket 4

File: BangazonAPI/Models/OrderProduct.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace BangazonAPI.Models
+{
+    public class OrderProduct
+    {
+
+        public int Id { get; set; }
+        public int OrderId { get; set; }
```
Comment (jordan-castelloe) [inline BangazonAPI/Models/OrderProduct.cs]: The user is never interacting directly with this table, so we don't need a model for it.

File: BangazonAPI/Models/TrainingProgram.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace BangazonAPI.Models
+{
+    public class TrainingProgram
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public DateTime StartDate { get; set; }
+        public DateTime EndDate { get; set; }
+        public int MaxAttendees { get; set; }
```
Comment (jordan-castelloe) [inline BangazonAPI/Models/TrainingProgram.cs:14]: From ticket 9:
`Employees who signed up for a training program should be included in the response`

We need a list of type `Employee` on here

**Review by jordan-castelloe — CHANGES_REQUESTED** [review]

File: BangazonAPI/Models/Customer.cs (hunk 1 above)
Comment (sydneywait) [reply BangazonAPI/Models/Customer.cs:12]: Added

File: BangazonAPI/Models/EmployeeTraining.cs (hunk 2 above)
Comment (sydneywait) [reply BangazonAPI/Models/EmployeeTraining.cs]: added

File: BangazonAPI/Models/Order.cs (hunk 3 above)
Comment (sydneywait) [reply BangazonAPI/Models/Order.cs:12]: added

File: BangazonAPI/Models/OrderProduct.cs (hunk 4 above)
Comment (sydneywait) [reply BangazonAPI/Models/OrderProduct.cs]: Deleted

File: BangazonAPI/Models/TrainingProgram.cs (hunk 5 above)
Comment (sydneywait) [reply BangazonAPI/Models/TrainingProgram.cs:14]: Added

File: BangazonAPI/Models/EmployeeTraining.cs (hunk 2 above)
Comment (jordan-castelloe) [reply BangazonAPI/Models/EmployeeTraining.cs]: So actually I don't think we need this model at all, based on the tickets. We could sub it out for just having a list of Employees on Training Programs and a list of Training Programs on Employees. Sorry if that was unclear!

**Comment by iamnickmitchell** [comment]
V.2 - 🍜

File: BangazonAPI/Models/EmployeeTraining.cs (hunk 2 above)
Comment (sydneywait) [reply BangazonAPI/Models/EmployeeTraining.cs]: Sorry, I read that as "I can think...." instead of "I can't think"...   Now deleted.

**Review by jordan-castelloe — APPROVED** [review]

**Pull request merged**